plugin.video.themoviedb.helper/resources/lib/service.py

```diff
@@ -0,0 +1,460 @@
+        self.radius = 20
+        self.addon_path = utils.makepath('special://profile/addon_data/plugin.video.themoviedb.helper/blur/')
+        self.home = xbmcgui.Window(10000)
+        self.image = artwork
+
+    def run(self):
+        if not self.image:
+            self.home.clearProperty('TMDbHelper.ListItem.BlurImage')
+            return
+        filename = '{}{}.png'.format(utils.md5hash(self.image), self.radius)
+        blurname = self.blur(self.image, filename)
+        self.home.setProperty('TMDbHelper.ListItem.BlurImage', blurname)
+
+    def blur(self, source, filename):
+        destination = self.addon_path + filename
+        try:
+            if xbmcvfs.exists(destination):
+                os.utime(destination, None)
+            else:
+                img = self.openimage(source, self.addon_path, filename)
+                img.thumbnail((256, 256))
+                img = img.convert('RGB')
+                img = img.filter(ImageFilter.GaussianBlur(self.radius))
+                img.save(destination)
+
+            return destination
+
+        except Exception:
+            return ''
+
+    def openimage(self, image, targetpath, filename):
+        """ Open image helper with thanks to sualfred """
+        # some paths require unquoting to get a valid cached thumb hash
+        cached_image_path = urllib.unquote(image.replace('image://', ''))
+        if cached_image_path.endswith('/'):
+            cached_image_path = cached_image_path[:-1]
+
+        cached_files = []
+        for path in [xbmc.getCacheThumbName(cached_image_path), xbmc.getCacheThumbName(image)]:
+            cached_files.append(os.path.join('special://profile/Thumbnails/', path[0], path[:-4] + '.jpg'))
+            cached_files.append(os.path.join('special://profile/Thumbnails/', path[0], path[:-4] + '.png'))
+            cached_files.append(os.path.join('special://profile/Thumbnails/Video/', path[0], path))
+
+        for i in range(1, 4):
+            try:
+                ''' Try to get cached image at first
+                '''
+                for cache in cached_files:
+                    if xbmcvfs.exists(cache):
+                        try:
+                            img = Image.open(xbmc.translatePath(cache))
+                            return img
+
+                        except Exception as error:
+                            utils.kodi_log('Image error: Could not open cached image --> %s' % error, 2)
+
+                ''' Skin images will be tried to be accessed directly. For all other ones
+                    the source will be copied to the addon_data folder to get access.
+                '''
+                if xbmc.skinHasImage(image):
+                    if not image.startswith('special://skin'):
+                        image = os.path.join('special://skin/media/', image)
+
+                    try:  # in case image is packed in textures.xbt
+                        img = Image.open(xbmc.translatePath(image))
```

Comment:
not closing the `Image` handler?

Reply:
Whoops! 
Closing here now:
https://github.com/xbmc/repo-plugins/blob/4f1a0b77afb48e4400e61c82fff9d7691a3a0b85/plugin.video.themoviedb.helper/resources/lib/service.py#L74